### PR TITLE
have cmake's build skip if it's already been done

### PR DESCRIPTION
**Pull request by komidore64**

### CODE REVIEW

File: subplays/cmake.yaml [hunk 1]
```diff
@@ -7,29 +7,51 @@
...
+    command: make install
+    args:
+      chdir: "{{ build_dir }}/cmake-{{ cmake_version }}"
+
+  - name: cmake build/install successfully
+    file:
+      path: "{{ build_dir}}/cmake-completed"
+      state: touch
+
+  when: cmake_completed.stat.islnk is not defined
```
Comment (ehelms) [inline subplays/cmake.yaml]: This `when` looks detached so I am not sure what it's being applied to.

Comment (komidore64) [reply subplays/cmake.yaml]: it's attached to the `block`

Comment (komidore64) [reply subplays/cmake.yaml]: I'm not opposed to putting the conditional immediately under the object it is being applied to, but from looking at Ansible docs, it appears the accepted format is placing the conditional at the bottom of the object.

Comment (ehelms) [reply subplays/cmake.yaml]: I agree with that, but its got a blank line above it so I can't tell if it applies to the entire set of plays or the task defined on line 52

Comment (ehelms) [reply subplays/cmake.yaml]: Err I think I see what's happening here. Man thats confusing and being stuck at the bottom makes it not obvious. Can it go at the top? This seems to rely on a fact that is generated part way through the play. Perhaps the play itself (the whole yaml file) shouldn't determine when it's run but rather the caller of it and let the play focus on just 'doing'. I don't know that there is a great answer.

Comment (komidore64) [reply subplays/cmake.yaml]: It's just a dictionary, so it should definitely be able to go at the top. I'll move the same conditional in `subplays/gcc.yaml` also.

Comment (komidore64) [reply subplays/cmake.yaml]: I was incorrect. Ansible complains about a syntax error :(

Comment (komidore64) [reply subplays/cmake.yaml]: I added a small comment to each block's `when` clause in hopes that the conditional's purpose is more clear.

**Pull request merged**